Index: core/Core.php
===================================================================
--- core/Core.php	(revision 115947)
+++ core/Core.php	(working copy)
@@ -82,9 +82,11 @@
 				$url = $_FILE_TO_URL_MAPPING[$testPath] 
 					. str_replace(DIRECTORY_SEPARATOR,'/',substr($fullPath,strlen($testPath)));
 				
-				$_SERVER['HTTP_HOST'] = parse_url($url, PHP_URL_HOST);
-				$_SERVER['SCRIPT_NAME'] = $_SERVER['PHP_SELF'] = parse_url($url, PHP_URL_PATH);
-				$_SERVER['REQUEST_PORT'] = parse_url($url, PHP_URL_PORT);
+				$components = parse_url($url);
+				$_SERVER['HTTP_HOST'] = $components['host'];
+				if( !empty($components['port']) ) $_SERVER['HTTP_HOST'] .= ':'.$components['port'];
+				$_SERVER['SCRIPT_NAME'] = $_SERVER['PHP_SELF'] = $components['path'];
+				if( !empty($components['port']) ) $_SERVER['REQUEST_PORT'] = $components['port'];
 			    break;
 			}
 			$testPath = dirname($testPath);
