Index: lang/en_GB.php
===================================================================
--- lang/en_GB.php	(revision 115975)
+++ lang/en_GB.php	(working copy)
@@ -188,7 +188,7 @@
 $lang['en_GB']['ForgotPasswordEmail.ss']['TEXT2'] = 'password reset link';
 $lang['en_GB']['ForgotPasswordEmail.ss']['TEXT3'] = 'for';
 $lang['en_GB']['Form']['DATENOTSET'] = '(No date set)';
-$lang['en_GB']['Form']['FIELDISREQUIRED'] = '%s is required.';
+$lang['en_GB']['Form']['FIELDISREQUIRED'] = '%s is required';
 $lang['en_GB']['Form']['LANGAOTHER'] = 'Other languages';
 $lang['en_GB']['Form']['LANGAVAIL'] = 'Available languages';
 $lang['en_GB']['Form']['NOTSET'] = '(not set)';
