Index: static-main.php
===================================================================
--- static-main.php	(revision 115975)
+++ static-main.php	(working copy)
@@ -34,19 +34,20 @@
 	// Define system paths (copied from Core.php)
 	if(!defined('BASE_PATH')) {
 		// Assuming that this file is sapphire/static-main.php we can then determine the base path
-		define('BASE_PATH', rtrim(dirname(dirname(__FILE__))), DIRECTORY_SEPARATOR);
+		define('BASE_PATH', str_replace(DIRECTORY_SEPARATOR, '/', rtrim(dirname(dirname(__FILE__)))), DIRECTORY_SEPARATOR);
 	}
 	if(!defined('BASE_URL')) {
+		$script = str_replace(DIRECTORY_SEPARATOR, '/', realpath($_SERVER['SCRIPT_FILENAME']));
 		// Determine the base URL by comparing SCRIPT_NAME to SCRIPT_FILENAME and getting the common elements
-		if(substr($_SERVER['SCRIPT_FILENAME'],0,strlen(BASE_PATH)) == BASE_PATH) {
-			$urlSegmentToRemove = substr($_SERVER['SCRIPT_FILENAME'],strlen(BASE_PATH));
+		if(substr($script,0,strlen(BASE_PATH)) == BASE_PATH) {
+			$urlSegmentToRemove = substr($script,strlen(BASE_PATH));
 			if(substr($_SERVER['SCRIPT_NAME'],-strlen($urlSegmentToRemove)) == $urlSegmentToRemove) {
 				$baseURL = substr($_SERVER['SCRIPT_NAME'], 0, -strlen($urlSegmentToRemove));
 				define('BASE_URL', rtrim($baseURL, DIRECTORY_SEPARATOR));
 			} 
 		}
 	}
-	
+		
 	$url = $_GET['url'];
 	// Remove base folders from the URL if webroot is hosted in a subfolder
 	if (substr(strtolower($url), 0, strlen(BASE_URL)) == strtolower(BASE_URL)) {
